README.md

```diff
@@ -56,60 +56,71 @@ NB_GRAPH_USERNAME=someuser
+| `NB_GRAPH_ROOT_CONT` | No                | Path to directory for graph databases in the _graph server container_                                                                    | `/var/opt/stardog` *                   | Docker                        |
+| `NB_GRAPH_PORT_HOST` | No                | Port number on the _host machine_ to map the graph server container port to                                                              | `5820`                                 | Docker                        |
+| `NB_GRAPH_PORT`      | No                | Port number used by the _graph server container_                                                                                         | `5820` *                               | Docker                        |
+
+_* These defaults should not be changed if using a Stardog backend._
+
+**NOTE:** To avoid conflicts related to [Docker's environment variable precedence](https://docs.docker.com/compose/environment-variables/envvars-precedence/), 
+ensure that any variables defined in your `.env` file are not already set in your current shell environment with **different** values.
+
+If needed, you can export all the variables defined in your `.env` file in one step by running the following:
 ```bash
 export $(cat .env | xargs)
 ```
 
-The below instructions for Docker and Python assume that you have at least set `NB_GRAPH_USERNAME` and `NB_GRAPH_PASSWORD` in your current environment.
+The below instructions for Docker and Python assume that you have at least set `NB_GRAPH_USERNAME` and `NB_GRAPH_PASSWORD` in your `.env`.
 
 ### Docker
 First, [install docker](https://docs.docker.com/get-docker/).
 
 You can then run a Docker container for the API in one of three ways:
-#### Option 1: Use the `docker-compose.yaml` file
+#### Option 1 (RECOMMENDED): Use the `docker-compose.yaml` file
 
 First, [install docker compose](https://docs.docker.com/compose/install/).
 
 If needed, update your `.env` file with optional environment variables for the docker compose configuration.
 
+**TIP:** Double check that the environment variables are resolved with your expected values using the command `docker compose config`.
+
 Use Docker Compose to spin up the containers by running the following in the repository root (where the `docker-compose.yml` file is):
 ```bash
 docker compose up -d
 ```
 
 #### Option 2: Use the latest image from Docker Hub
 ```bash
-export $(cat .env | xargs)  # export your environment variables 
+export $(cat .env | xargs)  # OPTIONAL: export your environment variables
 docker pull neurobagel/api
-docker run -d --name api -p 8000:8000 --env NB_GRAPH_USERNAME --env NB_GRAPH_PASSWORD neurobagel/api
+docker run -d --name=api -p ${NB_API_PORT_HOST:-8000}:${NB_API_PORT:-8000} --env-file=.env neurobagel/api
```

Comment:
I get what you have in mind here - but I don't think the README section for "minimal command to run this" is the right place for this relatively complex syntax. As far as I understand, the motivation for `NB_API_PORT_HOST` and `NB_API_PORT` is when I use `docker compose`, yes? in that case I'd just stick with a literal port value - we can still mention that this can be changed.